java/main/src/test/java/com/saucelabs/saucebindings/integration/VisualTest.java

```diff
@@ -0,0 +1,130 @@
+package com.saucelabs.saucebindings.integration;
+
+import com.saucelabs.saucebindings.GitManager;
+import com.saucelabs.saucebindings.SauceVisualException;
+import com.saucelabs.saucebindings.VisualResults;
+import com.saucelabs.saucebindings.VisualSession;
+import com.saucelabs.saucebindings.VisualSnapshot;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class VisualTest {
+    private VisualSession session;
+    private RemoteWebDriver driver;
+
+    @Before
+    public void setup() {
+        System.setProperty("BUILD_NAME", "Sauce Bindings");
+    }
+
+    @After
+    public void cleanUp() {
+        if (session != null) {
+            session.stop();
+        }
+    }
+
+    @Test
+    public void startsSession() {
+        session = new VisualSession("VisualTest startsSession");
+        driver = session.start();
+
+        assertNotNull(driver);
+        assertTrue(session.getSauceUrl().toString().contains("screener.io"));
+    }
+
+    @Test
+    public void getVisualResults() {
+        session = new VisualSession("VisualTest getVisualResults");
+        driver = session.start();
+
+        session.takeSnapshot("Blank");
+
+        VisualResults results = session.getVisualResults();
+
+        assertTrue(results.getPassed());
+        assertEquals("success", results.getStatus());
+
+        assertNull(results.getMessage());
+
+        assertEquals(Long.valueOf(1), results.getTotal());
+        assertEquals(Long.valueOf(1), results.getTotalAccepted());
+        assertEquals(Long.valueOf(0), results.getTotalNew());
+        assertEquals(Long.valueOf(0), results.getTotalChanged());
+        assertEquals(Long.valueOf(0), results.getTotalRejected());
+
+        List<VisualSnapshot> states = results.getSnapshots();
+        assertEquals(1, states.size());
+
+        VisualSnapshot snapshot = results.getSnapshots().get(0);
+        assertEquals("Blank", snapshot.getName());
+        assertEquals("VisualTest getVisualResults", snapshot.getGroupName());
+        assertEquals("accepted", snapshot.getStatus());
+        String url = snapshot.getUrl();
+        assertTrue(url.contains(GitManager.getCurrentBranch()));
```

Comment:
I prefer to use something like `Assert.assertThat(x, CoreMatchers.containsString("foo"));`. This is giving much better error messages with actual/expected values instead of just true/false